smac/facade/smac_facade.py

```diff
@@ -207,44 +253,64 @@ def __init__(self,
 
         # initial EPM
         types, bounds = get_types(scenario.cs, scenario.feature_array)
+        model_def_kwargs = {
+            'types':types,
+            'bounds':bounds,
+            'instance_features':scenario.feature_array,
+            'seed':rng.randint(MAXINT),
+            'pca_components':scenario.PCA_DIM,
+            'log_y':scenario.transform_y in ["LOG", "LOGS"],
+            'num_trees':scenario.rf_num_trees,
+            'do_bootstrapping':scenario.rf_do_bootstrapping,
+            'ratio_features':scenario.rf_ratio_features,
+            'min_samples_split':scenario.rf_min_samples_split,
+            'min_samples_leaf':scenario.rf_min_samples_leaf,
+            'max_depth':scenario.rf_max_depth
+            }
+        model_def_kwargs.update(model_kwargs)
         if model is None:
-            model = RandomForestWithInstances(types=types,
-                                              bounds=bounds,
-                                              instance_features=scenario.feature_array,
-                                              seed=rng.randint(MAXINT),
-                                              pca_components=scenario.PCA_DIM,
-                                              log_y=scenario.transform_y in ["LOG", "LOGS"],
-                                              num_trees=scenario.rf_num_trees,
-                                              do_bootstrapping=scenario.rf_do_bootstrapping,
-                                              ratio_features=scenario.rf_ratio_features,
-                                              min_samples_split=scenario.rf_min_samples_split,
-                                              min_samples_leaf=scenario.rf_min_samples_leaf,
-                                              max_depth=scenario.rf_max_depth)
+            model = RandomForestWithInstances(**model_def_kwargs)
+        elif inspect.isclass(model):
+            model = model(**model_def_kwargs)
+        #TODO since not all our models implement the AbstractEPM,
+        #no check of object done here
+            
         # initial acquisition function
+        acq_def_kwargs = {'model': model}
+        acq_def_kwargs.update(acquisition_function_kwargs)
```

Comment:
I think we should add a check here that model is not overwritten from the outside.

Reply:
why should we? I don't see a reason why someone should do that but I thought our code should be more flexible. So, I would not additional constraints.